From 8f3c09db4e27facbf0b82e3b601029b61fda86f3 Mon Sep 17 00:00:00 2001
From: Eli Collins <eli@apache.org>
Date: Thu, 20 Sep 2012 01:35:01 +0000
Subject: [PATCH 0631/1357] HDFS-3949. NameNodeRpcServer#join should join on both client and server RPC servers. Contributed by Eli Collins

git-svn-id: https://svn.apache.org/repos/asf/hadoop/common/branches/branch-2@1387839 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit ce71b6bd5b188538c62eeec2703acfd44a866ceb)

Conflicts:

	hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNodeRpcServer.java
---
 .../hdfs/server/namenode/NameNodeRpcServer.java    |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNodeRpcServer.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNodeRpcServer.java
index acfcbab..08c2fec 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNodeRpcServer.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNodeRpcServer.java
@@ -284,10 +284,13 @@ class NameNodeRpcServer implements NamenodeProtocols {
   }
   
   /**
-   * Wait until the RPC server has shut down.
+   * Wait until the RPC servers have shutdown.
    */
   void join() throws InterruptedException {
-    this.clientRpcServer.join();
+    clientRpcServer.join();
+    if (serviceRpcServer != null) {
+      serviceRpcServer.join();      
+    }
   }
   
   void stop() {
-- 
1.7.0.4

